File: backend/DAL/Entities/Notification.cs
using DAL.Entities;

namespace DAL;

public class Notification : Entity {
  public string Description { get; set; }
  public Uri? Url { get; set; }
  public DateTime? ReadAt { get; set; }
  public Guid UserId { get; set; }
  public Guid RouteId { get; set; }
  public Guid DangerId { get; set; }

  // Foreign Keys
  public User User { get; set; }
  public Route Route { get; set; }
  public Danger Danger { get; set; }
}

File: frontend/src/services/entities/Notification.ts
import { Danger } from "./Danger";
import { User } from "./User";
import { Route } from "./Route";


export interface Notification {
    id?: string;
    createdAt?: Date;
    updatedAt?: Date | undefined;
    description?: string | undefined;
    url?: string | undefined;
    readAt?: Date | undefined;
    userId?: string;
    routeId?: string;
    dangerId?: string;
    user?: User;
    route?: Route;
    danger?: Danger;
}
